routers/web/user/setting/account.go

```diff
@@ -233,14 +232,10 @@ func DeleteAccount(ctx *context.Context) {
 	ctx.Data["Title"] = ctx.Tr("settings")
 	ctx.Data["PageIsSettingsAccount"] = true
 
-	if _, _, err := auth.UserSignIn(ctx, ctx.Doer.Name, ctx.FormString("password")); err != nil {
-		if user_model.IsErrUserNotExist(err) {
-			loadAccountData(ctx)
+	if ctx.Doer.IsPasswordSet() && !ctx.Doer.ValidatePassword(ctx.FormString("password")) {
```

Comment:
Is it okay to use `ValidatePassword` rather than `UserSignIn` here, or does `UserSignIn` have different behavior?